src/Gaffer/ScriptNode.cpp

```diff
@@ -534,48 +534,42 @@ void ScriptNode::paste( Node *parent )
 	}
 }
 
+namespace
+{
+
 void handleReconnectionOfOutputPlug( const ScriptNode *script, const DependencyNode *dependencyNode, const Plug *outputPlug )
 {
-	// for plugs that hold output plugs we need to recurse
-	if(outputPlug->typeId() == Plug::staticTypeId())
+	const Plug *inPlug = dependencyNode->correspondingInput( outputPlug );
+	if ( inPlug )
 	{
-		for( OutputPlugIterator it( outputPlug ); !it.done(); ++it )
+		const Plug *srcPlug = inPlug->getInput<Plug>();
+		if ( srcPlug )
 		{
-			handleReconnectionOfOutputPlug( script, dependencyNode, it->get() );
+			// record this plug's current outputs, and reconnect them. This is a copy of outputPlug->outputs() rather
+			// than a reference, as reconnection can modify outputPlug->outputs()...
+			Plug::OutputContainer outputs = outputPlug->outputs();
+			for ( Plug::OutputContainer::const_iterator oIt = outputs.begin(); oIt != outputs.end(); ++oIt )
+			{
+				Plug *dstPlug = *oIt;
+				if ( dstPlug && dstPlug->acceptsInput( srcPlug ) && script->isAncestorOf( dstPlug ) )
+				{
+					dstPlug->setInput( const_cast<Gaffer::Plug*>(srcPlug) );
+					return;
```

Comment:
Oh yeah, that makes sense - I didn't know about the recursive iterator. Also, I guess I was under the impression that we'd have either a connection at the parent level or below and not both. Just out of curiosity, wouldn't a connection at the parent level mean that all children are connected as well (as in: a color plug connects all channels)? I will update the PR after the meeting. Thanks :)